pkgs/top-level/python-packages.nix

```diff
@@ -37,6 +37,20 @@ let
 
   graphiteVersion = "0.9.15";
 
+  fetchPypiMirror = ({ name ? null , pname ? null, version ? null, kind, ext, sha256}:
+    let
+      _name = if name == null then "${pname}-${version}" else name;
+      _pdrv = builtins.parseDrvName _name;
+    in
+      pkgs.fetchurl {
+        inherit sha256;
+        url = "mirror://pypiio/${kind}/${builtins.substring 0 1 _name}/${_pdrv.name}/${_name}${ext}";
```

Comment:
this actually won't work in all cases :) I know your code should work, but I found out that there are many different ways how URL is generated on pypi.python.org. also below code should

```nix
  fetchpypi = name: version: hash: pkgs.fetchurl {
      urls = [
        "https://pypi.python.org/packages/source/${builtins.substring 0 1 name}/${name}/${name}-${version}.tar.gz"
        "https://pypi.python.org/packages/source/${builtins.substring 0 1 name}/${name}/${pkgs.lib.toLower name}-${version}.tar.gz"
        "https://pypi.python.org/packages/source/${builtins.substring 0 1 name}/${name}/${name}-${version}.tgz"
        "https://pypi.python.org/packages/source/${builtins.substring 0 1 name}/${name}/${pkgs.lib.toLower name}-${version}.tgz"
        "https://pypi.python.org/packages/source/${builtins.substring 0 1 name}/${name}/${name}-${version}.zip"
        "https://pypi.python.org/packages/source/${builtins.substring 0 1 name}/${name}/${pkgs.lib.toLower name}-${version}.zip"
      ];
      md5 = "${hash}";
  };
```